hashintel/hash
# Refresh page during block development

## 🌟 What is the purpose of this PR?

This PR improves block DX by automatically reloading HASH EA when a block is changed. This PR is based on https://github.com/blockprotocol/blockprotocol/pull/317 and relies on the new `devReloadEndpoint` property in `block-metadata.json`.




## 🔗 Related links





- [Asana task](https://app.asana.com/0/1200211978612931/1202073961250686/f) _(internal)_
- https://github.com/blockprotocol/blockprotocol/pull/317





















## ⚠️ Known issues

Current implementation reloads the EA even if a locally developed block is not mounted (which should be rare). We should move this logic to a place where a block is mounted. This requires block metadata to be available there, which is not the case ATM.


## 🐾 Next steps

1.  Merge https://github.com/blockprotocol/blockprotocol/pull/317 and publish the new version of `block-scripts`
1.  Upgrade `block-scripts` version in this repo to unlock `yarn dev` improvement for the existing blocks


## 🛡 What tests cover this?



- manual + CI

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Run `yarn dev`
1.  Checkout https://github.com/blockprotocol/blockprotocol/pull/317 in another folder
1.  `cd packages/block-template && yarn dev`
1.  Add locally developed `block-template` block to the HASH app
1.  Change something in `block-template/app.tsx`
1.  Observe HASH app reloading

## 📹 Demo

https://user-images.githubusercontent.com/608862/171404245-1d8a7058-a9d4-4b51-a8e6-2ebaf87c5788.mp4
